### PR TITLE
[MSIEXEC][MSI] Sync msiexec to wine-9.8

### PR DESCRIPTION
**Pull request by MishaProductions**
## Purpose

Sync msiexec & msi to wine 9.8

JIRA issue: None

## Proposed changes

- Sync msiexec to wine-9.8
- Sync msi to wine-9.8

## Todo:
 - [x] Test

Before:
image
image
image

After:
image
image
image

### CODE REVIEW

File: media/doc/WINESYNC.txt [hunk 1]
```diff
@@ -254,7 +254,7 @@ base/applications/wordpad           # Synced to WineStaging-1.9.16
 base/applications/write             # Synced to WineStaging-3.3
 base/services/rpcss                 # Synced to WineStaging-3.3
 base/system/expand                  # Synced to WineStaging-3.3
-base/system/msiexec                 # Synced to WineStaging-3.3
+base/system/msiexec                 # Synced to Wine-9.7
```
Comment (SergeGautherie) [inline media/doc/WINESYNC.txt:257]: 9.7 or 9.8?

Comment (MishaProductions) [reply media/doc/WINESYNC.txt]: Fixed, it should be 9.8.

**Review by HBelusca — DISMISSED** [review]
When syncing a wine module, the corresponding winetest must also be synced.
Shouldn't this MSIEXEC sync go also with a sync of MSI.DLL ?

**Comment by MishaProductions** [comment]
> When syncing a wine module, the corresponding winetest must also be synced. Shouldn't this MSIEXEC sync go also with a sync of MSI.DLL ?

Done

**Comment by maj113** [comment]
While it doesn't relate to the current build errors you should also remove nonamless defines from the precomp header

File: media/doc/WINESYNC.txt [hunk 2]
```diff
@@ -116,7 +116,7 @@ dll/win32/msg711.acm          # Synced to WineStaging-4.18
 dll/win32/msgsm32.acm         # Synced to WineStaging-4.0
 dll/win32/mshtml              # Synced to WineStaging-1.7.55
 dll/win32/mshtml.tlb          # Synced to WineStaging-1.7.55
-dll/win32/msi                 # Synced to WineStaging-7.3 (+ dialog.c synced to bbce5d014db7f023b133d6d09e6846e027586f7d)
+dll/win32/msi                 # Synced to WineStaging-9.8
```
Comment (SergeGautherie) [inline media/doc/WINESYNC.txt:119]: WineStaging or Wine?

Comment (MishaProductions) [reply media/doc/WINESYNC.txt]: Fixed, it should be Wine.

**Review by cbialorucki — APPROVED** [review]
Thank you for also syncing the tests.

**Comment by binarymaster** [comment]
The final commit doesn't have full name of author 😕 

https://git.reactos.org/?p=reactos.git;a=commit;h=f4be6dc36fbb9fee4606493183f0a32a0a5e7135

**Comment by DarkFire01** [comment]
https://patch-diff.githubusercontent.com/raw/reactos/reactos/pull/6897.patch

**Pull request merged**